packages/substrate-wasm/src/index.ts

```diff
@@ -0,0 +1,525 @@
+      entity: 'contractEmitted',
+      conditions: [],
+    };
+    if (ds.processor?.options?.contract) {
+      queryEntry.conditions.push({field: 'contract', value: ds.processor.options.contract.toLowerCase()});
+    } else {
+      return;
+    }
+    if (filter?.from) {
+      queryEntry.conditions.push({field: 'from', value: filter.from});
+    }
+    // TODO, in order extract identifier will require abi/metadata
+    if (filter?.identifier) {
+      const eventIndex = await getEventIndex(filter.identifier, ds);
+      if (eventIndex === undefined) {
+        (global as any).logger?.warn(
+          `Unable to locate identifier ${filter?.identifier} in abi, will be omitted from dictionary filter`
+        );
+      } else {
+        queryEntry.conditions.push({field: 'eventIndex', value: eventIndex.toString()});
+      }
+    }
+    return queryEntry;
+  },
+};
+
+const CallProcessor: SecondLayerHandlerProcessor_1_0_0<
+  SubstrateHandlerKind.Call,
+  WasmCallFilter,
+  WasmCall,
+  AnyTuple, // TODO improve this type
+  WasmDatasource
+> = {
+  specVersion: '1.0.0',
+  baseFilter: [{module: 'contracts', method: 'call'}],
+  baseHandlerKind: SubstrateHandlerKind.Call,
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async transformer({ds, input: original}): Promise<[WasmCall]> {
+    const [dest, value, gasLimit, storageDepositLimit, data] = original.extrinsic.method.args;
+
+    const success = !original.events.find(
+      (evt) => evt.event.section === 'system' && evt.event.method === 'ExtrinsicFailed'
+    );
+    let decodedMessage: DecodedMessage | undefined;
+    try {
+      const iAbi = await buildAbi(ds);
+      decodedMessage = decodeMessage(data.toU8a(), iAbi);
+    } catch (e) {
+      // TODO setup ts config with global defs
+      (global as any).logger?.warn(`Unable to parse call arguments, will be omitted from result`);
+    }
+
+    const call: WasmCall = {
+      // Transaction properties
+      from: original.extrinsic.signer,
+      dest: dest as Address,
+      value: value as BalanceOf,
+      gasLimit: gasLimit as Weight,
+      storageDepositLimit: storageDepositLimit as Option<Compact<u128>>,
+      // if unable to decode, return as string
+      data: decodedMessage ? {args: decodedMessage?.args, message: decodedMessage.message} : data.toHex(),
+      success,
+      selector: getSelector(data.toU8a()),
+      // Transaction response properties
+      hash: original.extrinsic.hash.toHex(), // Substrate extrinsic hash
+      blockNumber: original.block.block.header.number.toNumber(),
+      blockHash: original.block.block.hash.toHex(), // Substrate block hash
+      timestamp: original.block.timestamp,
+    };
+    return [call];
+  },
+  async filterProcessor({ds, filter, input}): Promise<boolean> {
+    try {
+      const [dest, , , , data] = input.extrinsic.args;
+      const iAbi = await buildAbi(ds);
+      const decodeMessage = iAbi?.decodeMessage(data.toU8a());
```

Comment:
Could this be moved below the contract address check? Could speed things up if decoding is slow